src/borg/helpers/parseformat.py

```diff
@@ -271,16 +271,15 @@ def parse_file_size(s):
 
 
 def sizeof_fmt(num, suffix='B', units=None, power=None, sep='', precision=2, sign=False):
-    prefix = '+' if sign and num > 0 else ''
-
+    sign = '+' if sign else ''
+    fmt = '{0:{1}.{2}f}{3}{4}{5}'
+    prec = 0
     for unit in units[:-1]:
         if abs(round(num, precision)) < power:
-            if isinstance(num, int):
-                return "{}{}{}{}{}".format(prefix, num, sep, unit, suffix)
-            else:
-                return "{}{:3.{}f}{}{}{}".format(prefix, num, precision, sep, unit, suffix)
+            return fmt.format(num, sign, prec, sep, unit, suffix)
         num /= float(power)
-    return "{}{:.{}f}{}{}{}".format(prefix, num, precision, sep, units[-1], suffix)
+        prec = precision
+    return fmt.format(num, sign, prec, sep, units[-1], suffix)
```

Comment:
to avoid repeating the format code and also to have a single point where the function returns:
```
    fmt = '{0:{1}.{2}f}{3}{4}{5}'
    prec = 0
    for unit in units[:-1]:
        if abs(round(num, precision)) < power:
            break
        num /= float(power)
        prec = precision
    else:
        unit = units[-1]
    return fmt.format(num, sign, prec, sep, unit, suffix)
```

Reply:
Didn't know about for/else. Nice tweak for a single format statement!